# HG changeset patch
# User niparx <koliap@gmail.com>
# Date 1515160111 -7200
# Node ID 88f0d5be23c521e7504e3f6cb2a153428efcf159
# Parent  bc4fb04c1de686c23c6207eee69ae0ca9fddcb7c
[OGRE-592] RTShaderSystem compile error when using double precision (#622)

RTShaderSystem: compile error when using double precision

https://ogre3d.atlassian.net/browse/OGRE-592

diff --git a/Components/RTShaderSystem/src/OgreShaderFFPTexturing.cpp b/Components/RTShaderSystem/src/OgreShaderFFPTexturing.cpp
--- a/Components/RTShaderSystem/src/OgreShaderFFPTexturing.cpp
+++ b/Components/RTShaderSystem/src/OgreShaderFFPTexturing.cpp
@@ -654,8 +654,12 @@
             curFuncInvocation->pushOperand(ParameterFactory::createConstParam(Vector4(alphaValue)), Operand::OPS_IN);
         }
         else
-        {               
-            curFuncInvocation->pushOperand(ParameterFactory::createConstParam(Vector4(colourValue.ptr())), Operand::OPS_IN);
+        {
+            curFuncInvocation->pushOperand(ParameterFactory::createConstParam(Vector4((Real)colourValue.r,
+                                                                                      (Real)colourValue.g,
+                                                                                      (Real)colourValue.b,
+                                                                                      (Real)colourValue.a)),
+                                           Operand::OPS_IN);
         }
         
         curFuncInvocation->pushOperand(arg, Operand::OPS_OUT);
